dynamiqs/solvers/propagator.py

```diff
@@ -3,10 +3,18 @@
 from torch import Tensor
 
 from ..utils.progress_bar import tqdm
+from ..utils.td_tensor import ConstantTDTensor
 from .solver import AutogradSolver
 
 
 class Propagator(AutogradSolver):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        # check that Hamiltonian is time-independent
+        if not isinstance(self.H, ConstantTDTensor):
+            raise TypeError('Propagator solvers require time-independent Hamiltonians.')
```

Comment:
```suggestion
            raise TypeError('Propagator solvers require a time-independent Hamiltonian.')
```

Reply:
(maybe you used the `s` for the batching, but I think we should ignore the batching when referring to the Hamiltonian or state in written sentences, things are clearer this way)